src/block/Ice.php

```diff
@@ -51,7 +52,11 @@ public function ticksRandomly() : bool{
 
 	public function onRandomTick() : void{
 		if($this->position->getWorld()->getHighestAdjacentBlockLight($this->position->x, $this->position->y, $this->position->z) >= 12){
-			$this->position->getWorld()->useBreakOn($this->position);
+			$ev = new BlockMeltEvent($this, VanillaBlocks::WATER());
+			$ev->call();
+			if(!$ev->isCancelled()){
+				$this->position->getWorld()->setBlock($this->position, $ev->getNewState(), false);
```

Comment:
This shouldn't disable block updating (it's needed to recalculate light level and notify nearby entities).